src/main/java/picard/vcf/AccumulateVariantCallingMetrics.java

```diff
@@ -101,34 +104,22 @@ protected int doWork() {
                     throw new PicardException(String.format("Expected 1 row in the summary metrics file but saw %d", summaryMetricsFile.getMetrics().size()));
                 }
 
-                // re-calculate internal fields from derived fields and add it to the list of summary metrics
+                // re-calculate internal fields from derived fields and merge it into the main summary metrics
                 final CollectVariantCallingMetrics.VariantCallingSummaryMetrics summaryMetrics = summaryMetricsFile.getMetrics().get(0);
                 summaryMetrics.calculateFromDerivedFields(totalHetDepth);
-                summaries.add(summaryMetrics);
+                CollectVariantCallingMetrics.VariantCallingSummaryMetrics.foldInto(collapsedSummary, Collections.singleton(summaryMetrics));
```

Comment:
again, replace with 
```
collapsedSummary.merge(summaryMetrics)
```
and move the `calculateDerivedFields` to outside the loop.